### PR TITLE
Refactors the workflow of adding and getting time series

### PR DESCRIPTION
**Pull request by daniel-thom**

### CODE REVIEW

File: src/time_array_container.jl [hunk 1]
```diff
@@ -1,82 +0,0 @@
...
-    resolution::Dates.Period
-    internal::InfrastructureSystemsInternal
-
-    function TimeDataContainer(
-        data::SortedDict{Dates.DateTime, Array{T, N}},
-        resolution::Dates.Period,
-        internal::InfrastructureSystemsInternal,
-    ) where {T, N}
-        series_length = length(first(values(data)))
-        for (k, v) in data
```
Comment (jd-lara) [inline src/time_array_container.jl:12]: I understand the removal of TimeDataContainer but couldn't find where did these checks go.

**Review by jd-lara — APPROVED** [review]
LGTM, we should consider the checks on the array sizes.

File: src/descriptors/structs.json [hunk 2]
```diff
@@ -54,46 +54,65 @@
...
+        },
+        {
+          "name": "resolution",
+          "data_type": "Dates.Period",
+          "comment": "forecast resolution"
+        },
         {
           "name": "data",
-          "data_type": "SortedDict{Dates.DateTime, TimeSeries.TimeArray}",
+          "data_type": "SortedDict{Dates.DateTime, Array}",
```
Comment (jd-lara) [inline src/descriptors/structs.json:81]: Vector in the case of Deterministic?

File: src/time_array_container.jl (hunk 1 above)
Comment (daniel-thom) [reply src/time_array_container.jl:12]: Good catch.  fixed.

File: src/descriptors/structs.json (hunk 2 above)
Comment (daniel-thom) [reply src/descriptors/structs.json]: Fixed

**Pull request merged**